components/switch/switch.go

```diff
@@ -41,8 +41,8 @@ type Switch interface {
 	// GetPosition returns the current position of the switch.
 	GetPosition(ctx context.Context, extra map[string]interface{}) (uint32, error)
 
-	// GetNumberOfPositions returns the total number of valid positions for this switch.
-	GetNumberOfPositions(ctx context.Context, extra map[string]interface{}) (uint32, error)
+	// GetNumberOfPositions returns the total number of valid positions for this switch, along with their labels.
```

Comment:
`labels should either be nil, empty, or the same length has the number.`
can you add that?

Reply:
Should I add checks in the client/server code ensure this, and return nil for labels if there's a mismatch? Or perhaps error?